packages/reactivity/src/baseHandlers.ts

```diff
@@ -197,7 +197,14 @@ export const mutableHandlers: ProxyHandler<object> = {
 
 export const readonlyHandlers: ProxyHandler<object> = {
   get: readonlyGet,
-  set(target, key) {
+  set(target, key, value, receiver) {
+    // is computed()
+    if ((target as any).__v_isRef && (target as any).effect) {
+      // computed should be able to set its own private properties
+      if (key === '_dirty' || key === '_value') {
+        return set(target, key, value, receiver)
+      }
+    }
```

Comment:
Can we add `__v_isComputed` to `ComputedRefImpl`? like this:

```diff
class ComputedRefImpl<T> {
  private _value!: T
  private _dirty = true

  public readonly effect: ReactiveEffect<T>

  public readonly __v_isRef = true;
+  public readonly __v_isComputed = true;
  public readonly [ReactiveFlags.IS_READONLY]: boolean
}
```
I mean, this would be safer, consider the following scenario:

```js
const data = ref({ effect: true })
const rData = readonly(data)

rData._value = 'xxx' // Unexpectedly 
```

with the `__v_isComputed`, we can implement the `isComputed` function, just like `isRef`.

Reply:
I'm a bit confused by the `{effect: true}` - that shouldn't have an impact, as that would be in `target.value.effect`, wouldn't it?

But nonetheless propably a safer approach., I agree.

Reply:
Oh sure, but you know what i mean 😄